src/main/java/com/gsheet/student/service/GoogleAuthTokenProvider.java

```diff
@@ -1,23 +1,59 @@
 package com.gsheet.student.service;
 
+import com.google.auth.oauth2.AccessToken;
 import com.google.auth.oauth2.GoogleCredentials;
-import lombok.RequiredArgsConstructor;
-import lombok.SneakyThrows;
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.gsheet.student.exception.GoogleCredentialRefreshException;
 import org.springframework.core.io.Resource;
 
-@RequiredArgsConstructor
+import java.io.IOException;
+
 public class GoogleAuthTokenProvider implements AuthTokenProvider {
+
+    private static GoogleAuthTokenProvider instance;
+
+    private GoogleCredentials credentials;
```

Comment:
It is a bad idea to share credentials on class, as this service can be reusable and you will have a trables